web/src/emoji_picker.js

```diff
@@ -189,21 +188,22 @@ function refill_section_head_offsets($popover) {
 }
 
 export function reactions_popped() {
-    return $current_message_emoji_popover_elem !== undefined;
+    return emoji_popover_instance;
 }
 
 export function hide_emoji_popover() {
-    $(".has_popover").removeClass("has_popover has_emoji_popover");
+    current_message_id = null;
```

Comment:
Why is this line outside the `reactions_popped` check below?

I think maybe it would make sense to refactor this function to just start with:
```
if {!is_open()) {
    return;
}
```

since I believe all of the logic here is clearly not intended to run if the popover wasn't open in the first place.

Probably can be a follow-up commit.

Reply:
That refactoring helped to catch a bug in the user status picker. Made this as a follow-up commit.